.github/workflows/test.yaml

```diff
@@ -0,0 +1,87 @@
+name: 'Test'
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+
+  # Only checking latest Linux/UNIX on pulls to save execution times
+  test-main-pulls:
+    if: github.event_name == 'pull_request'
+    runs-on: "${{ matrix.os }}"
+    strategy:
+      matrix:
+        os:
+          - 'macos-latest'
+          - 'ubuntu-latest'
+    steps:
+      - uses: 'actions/checkout@v2'
+        with:
+          fetch-depth: 1
+      - name: 'Install Dependencies'
+        run: './test/install_deps.sh'
+      - name: 'Run all tests'
+        run: './test/run.sh'
+        shell: 'bash'
+      # - uses: 'docker/setup-buildx-action@v2'
```

Comment:
Do we need this comment? If we don't need let's just remove it.

Reply:
Ack. Done